boot/boot.go

```diff
@@ -327,3 +327,44 @@ func MarkBootSuccessful(dev Device) error {
 	}
 	return nil
 }
+
+var ErrSystemBootModeUnsupported = errors.New("system boot mode is unsupported")
+
+// bootSystemState is used for choosing the recovery system and mode for the
+// next boot.
+type bootSystemState interface {
+	// setSystemMode configures the bootloader to boot into the provided
+	// system in given mode.
+	setSystemMode(system, mode string) (bootStateUpdate, error)
+}
+
+// RecoverySystemInMode configures the bootloader to boot into the given
+// recovery system in a particular mode. When booting into a recovery system is
+// not supproted by the device, an ErrSystemBootModeUnsupported error is
+// returned.
+func RecoverySystemInMode(dev Device, systemLabel, mode string) error {
+	if dev == nil {
```

Comment:
do we do this check in other places?

Reply:
Hmm not really. Let me drop it in the next push.